safe/gui/widgets/dock.py

```diff
@@ -1488,178 +1501,180 @@ def print_map(self):
-                    'viewer on your system. The generated pdfs were saved '
-                    'as:')),
-                m.Paragraph(wrapped_map_path),
-                m.Paragraph(self.tr('and')),
-                m.Paragraph(wrapped_html_path))
-
-            # noinspection PyCallByClass,PyTypeChecker,PyTypeChecker
-            QtGui.QDesktopServices.openUrl(
-                QtCore.QUrl('file:///' + html_pdf_path,
-                            QtCore.QUrl.TolerantMode))
-            # noinspection PyCallByClass,PyTypeChecker,PyTypeChecker
-            QtGui.QDesktopServices.openUrl(
-                QtCore.QUrl('file:///' + map_pdf_path,
-                            QtCore.QUrl.TolerantMode))
-            self.show_dynamic_message(status)
+        :param map_report: Map Report instance that is ready to print
+        :type map_report: MapReport
+        """
+        map_report.setup_composition()
+
+        # Get Filename
+        map_title = map_report.map_title
+        if map_title is not None:
+            default_file_name = map_title + '.pdf'
+            default_file_name = default_file_name.replace(' ', '_')
         else:
-            # AG:
-            # See https://github.com/AIFDR/inasafe/issues/911
-            # We need to set the composition to the composer before loading
-            # the template
-            print_map.setup_composition()
-            self.composer = self.iface.createNewComposer()
-            self.composer.setComposition(print_map.composition)
-            print_map.load_template()
-
-            # Zoom to Full Extent
-            number_pages = print_map.composition.numPages()
-            if number_pages > 0:
-                height = \
-                    print_map.composition.paperHeight() * number_pages + \
-                    print_map.composition.spaceBetweenPages() * \
-                    (number_pages - 1)
-                self.composer.fitInView(
-                    0, 0,
-                    print_map.composition.paperWidth() + 1,
-                    height + 1,
-                    QtCore.Qt.KeepAspectRatio)
+            self.show_error_message(
+                self.tr('Keyword "map_title" not found.'))
+            return
 
-        self.hide_busy()
+        # Get output path
+        # noinspection PyCallByClass,PyTypeChecker
+        map_pdf_path = QtGui.QFileDialog.getSaveFileName(
+            self,
+            self.tr('Write to PDF'),
+            os.path.join(temp_dir(), default_file_name),
+            self.tr('Pdf File (*.pdf)'))
+        map_pdf_path = str(map_pdf_path)
+
+        if map_pdf_path is None or map_pdf_path == '':
+            self.show_dynamic_message(
+                m.Message(
+                    m.Heading(self.tr('Map Creator'), **WARNING_STYLE),
+                    m.Text(self.tr('Printing cancelled!'))))
+            return
+
+        # Print Impact Table
+        table_file_name = os.path.splitext(map_pdf_path)[0] + '_table.pdf'
```

Comment:
I should add that early versions also had issues which is why we didnt use it - I think 2.6 improved things to prevent text being cut off on pagination etc.